internal/providers/gitlab/repository_properties.go

```diff
@@ -0,0 +1,81 @@
+//
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitlab
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/xanzy/go-gitlab"
+
+	"github.com/stacklok/minder/internal/entities/properties"
+	provifv1 "github.com/stacklok/minder/pkg/providers/v1"
+)
+
+func (c *gitlabClient) getPropertiesForRepo(
+	ctx context.Context, getByProps *properties.Properties,
+) (*properties.Properties, error) {
+	uid, err := getByProps.GetProperty(properties.PropertyUpstreamID).AsString()
+	if err != nil {
+		return nil, fmt.Errorf("upstream ID not found or invalid: %w", err)
+	}
+
+	projectURLPath, err := url.JoinPath("projects", url.PathEscape(uid))
```

Comment:
I guess later we'll split this into a helper when we have more getProperties calls

Reply:
yep, all of this will get refactored in due time.